src/main/java/uk/gov/ons/registers/cdap/service/CHService.java

```diff
@@ -0,0 +1,131 @@
+        private static final Logger LOG = LoggerFactory.getLogger(CHdataHandler.class);
+
+        private Gson gson = new Gson();
+
+        @UseDataSet(Sic07.CH_DATASET_NAME)
+        private Table chData;
+
+        /**
+         * Returns The Business Information as a JSON object from an entered Company Number
+         */
+        @GET
+        @Path("/CH/number/{number}")
+        public void getBusinessByNumber(HttpServiceRequest request, HttpServiceResponder responder,
+                                        @PathParam("number") String number) {
+
+            Row chRow = chData.get(new Get(number));
+
+            //Error Handing of empty results
+            if (chRow.isEmpty()) {
+                LOG.debug("No record for Business with Company Number, {} found", number);
+                responder.sendStatus(HttpURLConnection.HTTP_NOT_FOUND);
+                return;
+            }
+
+            responder.sendJson(gson.toJsonTree(byteMapToJSON(chRow.getColumns())));
+
+        }
+
+        /**
+         * Returns The Businesses Information as a JSON objects Based on Post Code Area
+         */
+        @GET
+        @Path("/CH/postcodearea/{postcode}")
+        public void getBusinessByPostCodeArea(HttpServiceRequest request, HttpServiceResponder responder,
+                             @PathParam("postcode") String postcodeArea) {
+
+            ArrayList<JsonElement> jsonElementArrayList = new ArrayList<>();
+            Row row;
+
+            try (Scanner scanner = chData.scan(null, null)) {
+                while ((row = scanner.next()) != null) {
+                    String postCode = row.getString(POSTCODE_COLUMN);
+
+                    if (postCode != null) {
+                        String scanPostCode[] = postCode.split(" ");
+
+                        if (scanPostCode[0].equals(postcodeArea)){
+                            jsonElementArrayList.add(byteMapToJSON(row.getColumns()));
+                        }
+                    }
+                }
+            }
+
+            //Error Handing of empty results
+            if (jsonElementArrayList.isEmpty()) {
+                LOG.debug("No records found for Businesses in PostCode Area: {}", postcodeArea);
+                responder.sendStatus(HttpURLConnection.HTTP_NOT_FOUND);
+                return;
+            }
+
+            responder.sendJson(gson.toJsonTree(jsonElementArrayList));
+
+        }
+
+        /**
+         * Method that takes byte[] HashMap from results and returns a decoded JSON object with the results
+         */
+        private JsonElement byteMapToJSON(Map<byte[], byte[]> hashMap){
+
+            HashMap<String, String> chBusinessData = new HashMap<>();
+
+            // Iterates thought results and casts the byte[] objects to Strings to a <String, String> HashMap
+            for (Map.Entry<byte[], byte[]> entry : hashMap.entrySet()) {
+                String keyString = new String(entry.getKey());
```

Comment:
Using the platform default String encoding here which is risky.  Best practice is to explicitly specify the charset.